rust/azure_iot_operations_connector/src/data_transformer.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Traits, types, and implementations for Azure IoT Operations Connector Data Transformers.
+
+#![allow(missing_docs)]
+
+use std::sync::Arc;
+
+use azure_iot_operations_mqtt::interface::CompletionToken;
+use azure_iot_operations_services::azure_device_registry::{AssetDataset, AssetDatasetDataPoint};
+
+use crate::destination_endpoint::Forwarder;
+use crate::{Data, base_connector::managed_azure_device_registry::Reporter};
+
+pub trait DataTransformer {
```

Comment:
Possibly outside the scope of this PR, but is there any way to share data between instances of a data-transformer with the current setup? If we have multiple WASM transformers, it'd probably benefit us to share the engine at least to reduce overhead.

Reply:
If it's record-based data, Arc would allow is to share information. For something more complex, not sure, would need to know more about what you'd like to see shared.

Reply:
kinda addressed in a response to your other comment, I'll add something for this PR to address this

Reply:
In this case, it's the entire WASM engine. So a bit more complex than record-based data. It's designed to be thread-safe (or at least [the underlying engine is](https://docs.rs/wasmtime/32.0.0/wasmtime/struct.Engine.html); the TK wrapper is arc-mutex-ed, so there might be some contention there). It's effectively designed to be a per-process singleton.

Reply:
updated to having DataTransformer passed into connector::new() that is then able to create the DatasetDataTransformers. Added a lot of generics, but we were going to need those to be plumbed through anyways in the next PR